curriculum/challenges/english/07-scientific-computing-with-python/scientific-computing-with-python-projects/budget-app.md

```diff
@@ -74,34 +65,592 @@ Percentage spent by category
+t = main(module='test_module', exit=False)
+t.result.wasSuccessful()
+`;
+    const out = __pyodide.runPython(testCode);
+    assert(out);
+  }
+})
+```
+
+Expected `withdraw` method to return `False`.
+
+```js
+({
+  test: () => {
+    pyodide.FS.writeFile('/home/pyodide/budget.py', code);
+    pyodide.FS.writeFile('/home/pyodide/test_module.py',`
+import unittest
+import budget
+from importlib import reload
+
+reload(budget)
+class UnitTests(unittest.TestCase):
+    maxDiff = None
+    def setUp(self):
+        self.food = budget.Category("Food")
+    
+    def test_withdraw_no_funds(self):
+        self.food.deposit(100, "deposit")
+        good_withdraw = self.food.withdraw(100.10)
+        self.assertEqual(good_withdraw, False, 'Expected "withdraw" method to return "False".')    
+`);
+    const testCode = `
+from unittest import main
+from importlib import reload
+import test_module
+
+reload(test_module)
+t = main(module='test_module', exit=False)
+t.result.wasSuccessful()
+`;
+    const out = __pyodide.runPython(testCode);
+    assert(out);
+  }
+})
+```
+
+Expected `transfer` method to return `False`.
+
+```js
+({
+  test: () => {
+    pyodide.FS.writeFile('/home/pyodide/budget.py', code);
+    pyodide.FS.writeFile('/home/pyodide/test_module.py',`
+import unittest
+import budget
+from importlib import reload
+
+reload(budget)
+class UnitTests(unittest.TestCase):
+    maxDiff = None
+    def setUp(self):
+        self.food = budget.Category("Food")
+        self.entertainment = budget.Category("Entertainment")
+    
+    def test_transfer_no_funds(self):
+        self.food.deposit(100, "deposit")
+        good_transfer = self.food.transfer(200, self.entertainment)
+        self.assertEqual(good_transfer, False, 'Expected "transfer" method to return "False".')    
+`);
+    const testCode = `
+from unittest import main
+from importlib import reload
+import test_module
+
+reload(test_module)
+t = main(module='test_module', exit=False)
+t.result.wasSuccessful()
+`;
+    const out = __pyodide.runPython(testCode);
+    assert(out);
+  }
+})
+```
+
+Printing `Category("Food")` should give a different string representation of the object.
+
+```js
+({
+  test: () => {
+    pyodide.FS.writeFile('/home/pyodide/budget.py', code);
+    pyodide.FS.writeFile('/home/pyodide/test_module.py',`
+import unittest
+import budget
+from importlib import reload
+
+reload(budget)
+class UnitTests(unittest.TestCase):
+    maxDiff = None
+    def setUp(self):
+        self.food = budget.Category("Food")
+        self.entertainment = budget.Category("Entertainment")
+    
+    def test_to_string(self):
+        self.food.deposit(900, "deposit")
+        self.food.withdraw(45.67, "milk, cereal, eggs, bacon, bread")
+        self.food.transfer(20, self.entertainment)
+        actual = str(self.food)
+        expected = f"*************Food*************\ndeposit                 900.00\nmilk, cereal, eggs, bac -45.67\nTransfer to Entertainme -20.00\nTotal: 834.33"
```

Comment:
```suggestion
        expected = "*************Food*************\\ndeposit                 900.00\\nmilk, cereal, eggs, bac -45.67\\nTransfer to Entertainme -20.00\\nTotal: 834.33"
```
The `\`s need escaping, otherwise JavaScript interprets `\n` as being the newline character. We don't want that, we want Python to get the literal characters `\` and `n`. Then Python will correctly interpret them as newlines.